modules/gapi/src/api/render_ocv.cpp

```diff
@@ -27,29 +28,51 @@ inline void mosaic(cv::Mat& mat, const cv::Rect &rect, int cellSz)
     }
 };
 
-inline void image(cv::Mat& mat,
-                  const cv::Point& org,
-                  const cv::Mat& img,
-                  const cv::Mat& alpha)
+inline void blendImage(const cv::Mat& img,
+                       const cv::Mat& alpha,
+                       const cv::Point& org,
+                       cv::Mat background)
 {
-    auto roi = mat(cv::Rect(org, img.size()));
+    GAPI_Assert(alpha.type() == CV_32FC1);
+    GAPI_Assert(background.channels() == 3u);
+
+    cv::Mat roi = background(cv::Rect(org, img.size()));
     cv::Mat img32f_w;
     cv::merge(std::vector<cv::Mat>(3, alpha), img32f_w);
 
     cv::Mat roi32f_w(roi.size(), CV_32FC3, cv::Scalar::all(1.0));
     roi32f_w -= img32f_w;
 
     cv::Mat img32f, roi32f;
+    if (img.type() == CV_32FC3) {
+        img.copyTo(img32f);
```

Comment:
Why use copy in this case? Can't we just say that `img32f` _is_ `img` (i.e. simply assign?)

Reply:
Because we change `img32f`

Reply:
Ok, got iy